Session4_v2.Rmd

```diff
@@ -0,0 +1,354 @@
+
+Then we use the `ggplot` function to visualise the buildings by age. The specific function to represent information as a map is `geom_sf()`. The rest works like other graphs and visualisation, with `aes()` for the aesthetics.
+
+```{r map-age}
+buildings$build_date <- as.numeric(if_else(as.numeric(buildings$start_date) <1900, 1900,as.numeric(buildings$start_date)))
+
+ ggplot(data = buildings) +
+   geom_sf(aes(fill = build_date, colour=build_date))  +
+   scale_fill_viridis_c(option = "viridis")+
+   scale_colour_viridis_c(option = "viridis")
+```
+
+So this reveals the historical centre of [city] and the various extensions.
+Anything odd? what? around the centre? Why these limits / isolated points?
+
+## Challenge: import an interactive basemap layer under the buildings with `Leaflet`
+
+Some information here: [leaflet package documentation](https://rstudio.github.io/leaflet/)
+---
+
+## Solution
+
+```{r, echo=FALSE}
+buildings2 <- buildings %>%
+  st_transform(.,crs=4326)
+
+leaflet(buildings2) %>%
+# addTiles()
+  addProviderTiles(providers$CartoDB.Positron) %>%
+  addPolygons(color = "#444444", weight = 0.1, smoothFactor = 0.5,
+  opacity = 0.2, fillOpacity = 0.8,
+  fillColor = ~colorQuantile("YlGnBu", -build_date)(-build_date),
+  highlightOptions = highlightOptions(color = "white", weight = 2,
+    bringToFront = TRUE))
+
+```
+
+
+# Part 2. Basic GIS operations with the `sf` package
+
+
+## Why `sf`?
+
+`sf` is a package which supports simple features (sf), ["a standardized way to
+encode spatial vector data."](https://cran.r-project.org/web/packages/sf/sf.pdf).
+It contains a large set of functions to achieve all the operations on vector spatial data for which you might use traditional GIS software: change the coordinate system, join layers, intersect or unit polygons, create buffers and centroids, etc. cf. the `sf` [cheatsheet](https://raw.githubusercontent.com/rstudio/cheatsheets/main/sf.pdf).
+
+
+
+## Conservation in Brielle, NL
+
+Let's focus on old building and imagine we're in charge of their conservation. We want to know how much of the city would be affected by a non-construction zone of 100m around pre-1850 buildings.
+
+Let's select them and see where they are.
+
+
+```{r conservation}
+old <- 1800 # year prior to which you consider a building old
+
+summary(buildings$start_date)
+buildings$start_date <- as.numeric(as.character(buildings$start_date))
+
+old_buildings <- buildings %>%
+  filter(start_date <= old)
+
+ ggplot(data = old_buildings) + geom_sf(colour="red")
+ 
+```
+As conservationists, we want to create a zone around historical buildings where building regulation will have special restrictions to preserve historical buildings. 
+
+## Buffer
+
+Let's say this zone should be 500 meters. In GIS terms, we want to create a buffer around polygons. The corresponding function `sf` is `st_buffer`, with 2 arguments: the polygons around which to create buffers, and the radius of the buffer.
```

Comment:
I'm a bit confused about 500 meters. Is it the measurement of a radius around each building?

Reply:
Yeah, sorry that was from the past version. I've updated to 100m everywhere now.